subxt/src/backend/legacy/rpc_methods.rs

```diff
@@ -0,0 +1,532 @@
+/// some `T: Config` trait which determines some of the types that the RPC methods will
+/// take or hand back.
+pub struct LegacyRpcMethods<T> {
+    client: RpcClient,
+    _marker: std::marker::PhantomData<T>,
+}
+
+impl<T> Clone for LegacyRpcMethods<T> {
+    fn clone(&self) -> Self {
+        Self {
+            client: self.client.clone(),
+            _marker: self._marker,
+        }
+    }
+}
+
+impl<T> std::fmt::Debug for LegacyRpcMethods<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("LegacyRpcMethods")
+            .field("client", &self.client)
+            .field("_marker", &self._marker)
+            .finish()
+    }
+}
+
+impl<T: Config> LegacyRpcMethods<T> {
+    /// Instantiate the legacy RPC method interface.
+    pub fn new(client: RpcClient) -> Self {
+        LegacyRpcMethods {
+            client,
+            _marker: std::marker::PhantomData,
+        }
+    }
+
+    /// Fetch the raw bytes for a given storage key
+    pub async fn state_get_storage(
+        &self,
+        key: &[u8],
+        hash: Option<T::Hash>,
+    ) -> Result<Option<StorageKey>, Error> {
+        let params = rpc_params![to_hex(key), hash];
+        let data: Option<Bytes> = self.client.request("state_getStorage", params).await?;
+        Ok(data.map(|b| b.0))
+    }
+
+    /// Returns the keys with prefix with pagination support.
+    /// Up to `count` keys will be returned.
+    /// If `start_key` is passed, return next keys in storage in lexicographic order.
+    pub async fn state_get_keys_paged(
+        &self,
+        key: &[u8],
+        count: u32,
+        start_key: Option<&[u8]>,
+        at: Option<T::Hash>,
+    ) -> Result<Vec<StorageData>, Error> {
+        let start_key = start_key.map(to_hex);
+        let params = rpc_params![to_hex(key), count, start_key, at];
+        let data: Vec<Bytes> = self.client.request("state_getKeysPaged", params).await?;
+        Ok(data.into_iter().map(|b| b.0).collect())
+    }
+
+    /// Fetch the genesis hash
+    pub async fn genesis_hash(&self) -> Result<T::Hash, Error> {
+        let block_zero = 0u32;
+        let params = rpc_params![block_zero];
+        let genesis_hash: Option<T::Hash> =
+            self.client.request("chain_getBlockHash", params).await?;
+        genesis_hash.ok_or_else(|| "Genesis hash not found".into())
+    }
+
+    /// Fetch the metadata via the legacy `state_getMetadata` RPC method.
+    pub async fn state_get_metadata(&self, at: Option<T::Hash>) -> Result<Metadata, Error> {
+        let bytes: Bytes = self
+            .client
+            .request("state_getMetadata", rpc_params![at])
+            .await?;
+        let metadata = Metadata::decode(&mut &bytes[..])?;
+        Ok(metadata)
+    }
+
+    /// Fetch system health
+    pub async fn system_health(&self) -> Result<SystemHealth, Error> {
+        self.client.request("system_health", rpc_params![]).await
```

Comment:
I did not know these RPC methods were available, maybe we should add an example, that showcases all the little things we can fetch from a node?